Source/Tools/LinuxEmulation/LinuxSyscalls/Syscalls.cpp

```diff
@@ -759,6 +787,91 @@ uint32_t SyscallHandler::CalculateGuestKernelVersion() {
 }
 
 uint64_t SyscallHandler::HandleSyscall(FEXCore::Core::CpuStateFrame* Frame, FEXCore::HLE::SyscallArguments* Args) {
+  // Need to reconstruct RIP from the return address. Which will be inside the FEX JIT.
+  const uint64_t RIP =
```

Comment:
This handling that was added to this function should be moved in to the `ExecuteFilter` and `RestoreRIPFromHostPC` only called if RIP is actually needed. To keep seccomp handling consolidated in the seccomp handler and reduce overhead for when seccomp isn't enabled.